addons/account/models/account_partial_reconcile.py

```diff
@@ -135,9 +136,25 @@ def unlink(self):
     @api.model_create_multi
     def create(self, vals_list):
         partials = super().create(vals_list)
+        self._pay_matched_payment(partials)
         self._update_matching_number(partials.debit_move_id + partials.credit_move_id)
         return partials
 
+    @api.model
+    def _pay_matched_payment(self, partials):
+        for partial in partials:
+            matched_payments = (partial.credit_move_id | partial.debit_move_id).move_id.matched_payment_ids
+            amount = 0
+            for matched_payment in matched_payments:
+                if matched_payment.payment_type == 'inbound':
+                    amount = partial.debit_amount_currency
```

Comment:
@william-andre I tried creating  a test case where the `credit_amount_currency` != `debit_amount_currency` but failed to do it. Can you help me with more details on how to create it? :)

Reply:
You need an invoice in one foreign currency, and the payment transaction in another currency

Reply:
I tried but still the `debit_amount_currency = credit_amount_currency` still holds

Reply:
A quick search on `debit_amount_currency` gives you some results where the values are different, i.e. `test_reconcile_exchange_difference_on_partial_one_debit_foreign_currency_debit_expense_partial_payment` ( :scroll: :scroll: :scroll: )